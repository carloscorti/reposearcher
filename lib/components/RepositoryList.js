import React from 'react';

// import { connect } from 'react-redux';
// import * as actions from '../actions';

import Repository from './Repository';

const RepositoryList = (props) => {
  // console.log(props);

  const renderContent = (repos, welcome, filteredRepos) => {
    switch (repos) {
      case null:
        return (
          <>
            {/* <nav>
              <ul className="repo-nav">
                <li>Repositories</li>
              </ul>
            </nav> */}
            <section className="repositories">
              <div className="pinned-repo-wrapper">
                <div className="pinned-repo">
                  <h3>{welcome}</h3>
                  {/* <p>{description}</p>
                  <ul className="stats">
                    <li>
                      <span className="icon">ðŸ”µ</span>
                      <span>{language}</span>
                    </li>
                  </ul> */}
                </div>
              </div>
            </section>
          </>
        );
      default:
        return (
          <>
            {/* <nav>
              <ul className="repo-nav">
                <li>Repositories {filteredRepos.length}</li>
              </ul>
            </nav> */}
            <section className="repositories">
              {repos.length !== Object.values(filteredRepos).length && (
                <div className="pinned-repo-wrapper match">
                  <div className="pinned-repo">
                    <p>
                      <b>{Object.values(filteredRepos).length}</b> repositories
                      matching <b>{props.search}</b>
                    </p>
                  </div>
                </div>
              )}
              {Object.values(filteredRepos).map((repo) => (
                <Repository key={repo.id} data={repo} />
              ))}
            </section>
          </>
        );
    }
  };

  return <>{renderContent(props.repos, props.welcome, props.filteredRepos)}</>;
};

// const mapStatetoProps = ({ repos, welcome }) => {
//   return { repos, welcome };
// };

// export default connect(mapStatetoProps)(RepositoryList);

export default RepositoryList;
