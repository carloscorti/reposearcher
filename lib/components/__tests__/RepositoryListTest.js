import React from 'react';
import RepositoryList from '../RepositoryList';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('RepositoryList', () => {
  const testProps = {
    filteredRepos: [
      {
        id: 1,
        data: {
          name: 'n1',
          description: 'd1',
          language: 'l1',
          updated_at: 'ud1',
          html_url: 'url1',
        },
      },
      {
        id: 2,
        data: {
          name: 'n2',
          description: 'd2',
          language: 'l2',
          updated_at: 'ud2',
          html_url: 'url2',
        },
      },
    ],
    search: 'testSearch',
    welcome: 'testWelcome',
    repos: [{}, {}, {}],
  };

  const testPropsNoMatch = {
    filteredRepos: [
      { id: 1, data: {} },
      { id: 2, data: {} },
      { id: 3, data: {} },
    ],
    search: 'testSearchNoMatch',
    welcome: 'testWelcomeNoMatch',
    repos: [{}, {}, {}],
  };

  const testPropsNullRepos = {
    filteredRepos: [
      { id: 1, data: {} },
      { id: 2, data: {} },
      { id: 3, data: {} },
    ],
    search: 'testSearchNullRepos',
    welcome: 'testWelcomeNullRepos',
    repos: null,
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <RepositoryList
        filteredRepos={testProps.filteredRepos}
        search={testProps.search}
        welcome={testProps.welcome}
        repos={testProps.repos}
      />
    );

    expect(wrapper.find('Repository').length).toBe(2);
    expect(wrapper.find('b').length).not.toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly filtered repos msg', () => {
    const wrapper = shallow(
      <RepositoryList
        filteredRepos={testPropsNoMatch.filteredRepos}
        search={testPropsNoMatch.search}
        welcome={testPropsNoMatch.welcome}
        repos={testPropsNoMatch.repos}
      />
    );
    expect(wrapper.find('b').length).toBe(0);
  });

  it('renders correctly case null repos', () => {
    const wrapper = shallow(
      <RepositoryList
        filteredRepos={testPropsNullRepos.filteredRepos}
        search={testPropsNullRepos.search}
        welcome={testPropsNullRepos.welcome}
        repos={testPropsNullRepos.repos}
      />
    );

    expect(wrapper.find('h3').text()).toEqual(testPropsNullRepos.welcome);
    expect(wrapper.find('Repository').length).toBe(0);
  });
});
