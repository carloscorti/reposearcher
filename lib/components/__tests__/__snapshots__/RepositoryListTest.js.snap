// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryList renders correctly 1`] = `
<Fragment>
  <section
    className="repositories"
  >
    <div
      className="pinned-repo-wrapper match"
    >
      <div
        className="pinned-repo"
      >
        <p>
          <b>
            2
          </b>
           repositories matching
           
          <b>
            testSearch
          </b>
        </p>
      </div>
    </div>
    <Repository
      data={
        Object {
          "data": Object {
            "description": "d1",
            "html_url": "url1",
            "language": "l1",
            "name": "n1",
            "updated_at": "ud1",
          },
          "id": 1,
        }
      }
      key="1"
    />
    <Repository
      data={
        Object {
          "data": Object {
            "description": "d2",
            "html_url": "url2",
            "language": "l2",
            "name": "n2",
            "updated_at": "ud2",
          },
          "id": 2,
        }
      }
      key="2"
    />
  </section>
</Fragment>
`;
